src/legacy/core_plugins/data/public/index.ts

```diff
@@ -25,25 +34,34 @@ class DataPlugin {
   private readonly indexPatterns: IndexPatternsService;
   private readonly search: SearchService;
   private readonly query: QueryService;
+  private readonly expressions: ExpressionsService;
 
   constructor() {
     this.indexPatterns = new IndexPatternsService();
     this.query = new QueryService();
     this.search = new SearchService();
+    this.expressions = new ExpressionsService();
   }
 
-  public setup() {
+  public setup(): DataSetup {
```

Comment:
@lukeelmers could setup function be async ? then we could expose interpreter in our contract instead of promise which resolves with interpreter.

Reply:
It can be async, but it will slow down loading of all plugins.

Reply:
lets leave it for now, most likely we can refactor interpreter to not need async initialization. atm it only needs it to get the list of server functions.

Reply:
Left a TODO in the code